frontend/src/components/Semester/SemesterList/SemesterList.js

```diff
@@ -0,0 +1,193 @@
+import { COMMON_GROUP_TITLE } from '../../../constants/translationLabels/common';
+import { search } from '../../../helper/search';
+import { getGroupsOptionsForSelect } from '../../../utils/selectUtils';
+import SemesterCopyForm from '../../../containers/SemesterPage/SemesterCopyForm';
+import SemesterCard from './SemesterCard';
+import CustomDialog from '../../../containers/Dialogs/CustomDialog';
+import { MultiselectForGroups } from '../../../helper/MultiselectForGroups';
+import i18n from '../../../i18n';
+
+const SemesterList = (props) => {
+    const { t } = useTranslation('formElements');
+
+    const {
+        archived,
+        disabled,
+        term,
+        semesters,
+        selectSemesterSuccess,
+        createArchivedSemester,
+        setOpenConfirmDialog,
+        removeSemesterCard,
+        setDefaultSemesterById,
+        isOpenConfirmDialog,
+        semesterCopy,
+        options,
+        setOpenSuccessSnackbar,
+        setGroupsToSemester,
+        toggleSemesterVisibility,
+        loading,
+        // archivedSemesters,
+        // getArchivedSemesterById,
+    } = props;
+    const [isOpenSemesterCopyForm, setIsOpenSemesterCopyForm] = useState(false);
+    const [confirmDialogType, setConfirmDialogType] = useState('');
+    const [isOpenGroupsDialog, setIsOpenGroupsDialog] = useState(false);
+
+    const [semesterGroupsOptions, setSemesterGroupsOptions] = useState([]);
+    const [semesterId, setSemesterId] = useState(null);
+
+    const searchArr = ['year', 'description', 'startDay', 'endDay'];
+
+    const visibleItems = search(semesters, term, searchArr);
+
+    const showConfirmDialog = (id, dialogType) => {
+        setSemesterId(id);
+        setConfirmDialogType(dialogType);
+        setOpenConfirmDialog(true);
+    };
+    const submitSemesterCopy = ({ toSemesterId }) => {
+        semesterCopy({
+            fromSemesterId: semesterId,
+            toSemesterId,
+        });
+        setIsOpenSemesterCopyForm(false);
+    };
+    const closeSemesterCopyForm = () => {
+        setIsOpenSemesterCopyForm(false);
+    };
+
+    const acceptConfirmDialog = (currentSemesterId) => {
+        setOpenConfirmDialog(false);
+        const foundSemester = semesters.find(
+            (semesterItem) => semesterItem.id === currentSemesterId,
+        );
+        const isDisabled = disabled;
+        if (confirmDialogType === dialogTypes.SET_DEFAULT) {
+            setDefaultSemesterById(currentSemesterId, isDisabled);
+        } else if (confirmDialogType === dialogTypes.SET_VISIBILITY_ENABLED) {
+            toggleSemesterVisibility({
+                ...foundSemester,
+                disable: false,
+            });
+        } else if (confirmDialogType === dialogTypes.SET_VISIBILITY_DISABLED) {
+            toggleSemesterVisibility({
+                ...foundSemester,
+                disable: true,
+            });
+        } else {
+            removeSemesterCard(currentSemesterId);
+        }
+    };
```

Comment:
In this case, we can use the switch case

Reply:
done